clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -2,47 +2,110 @@ import { h, FunctionComponent } from "preact";
 import { useState, useCallback, useMemo } from "preact/hooks";
 import ConsentBanner from "../ConsentBanner";
 
-import { debugLog, hasActionNeededNotices } from "../../lib/consent-utils";
+import {
+  debugLog,
+  hasActionNeededNotices,
+  transformUserPreferenceToBoolean,
+} from "../../lib/consent-utils";
 
 import "../fides.css";
 import Overlay from "../Overlay";
-import { TcfConsentButtons } from "../ConsentButtons";
+import { TcfConsentButtons } from "./TcfConsentButtons";
 import { OverlayProps } from "../types";
 import TcfTabs from "./TcfTabs";
+import {
+  TCFFeatureRecord,
+  TCFPurposeRecord,
+  TCFVendorRecord,
+} from "../../lib/tcf/types";
+
+const resolveConsentValueFromTcfModel = (
```

Comment:
this is the kind of thing that can be taken out of this component and put in `lib` instead if we need to initialize the cookie value before rendering the overlay https://github.com/ethyca/fides/issues/3886